scripts/vagrant_lock.sh

```diff
@@ -0,0 +1,65 @@
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##########################################################################
+
+# This file should not be invoked directly, but sourced from other sh scripts.
+
+# Creates a lockfile to help with new user confusion when launching a vagrant
+# vm. See https://github.com/oppia/oppia/pull/2749 for details.
+#
+# It can be overridden by passing --nolock to start.sh
+
+
+# This keeps all the arguments passed in, except for --nolock just in case
```

Comment:
Stripping the nolock seems a bit complicated. Could we maybe have a flag less likely to have problems downstream, like ``novagrantlock`` to avoid the extra complexity? Bash can get complicated quickly, which makes it harder to maintain in the future.